pkg/minikube/assets/addons.go

```diff
@@ -762,36 +762,32 @@ func parseMapString(str string) map[string]string {
 	return mapResult
 }
 
-// mergeMaps creates a map with the union of `sourceMap` and `overrideMap` where collisions take the value of `overrideMap`.
-func mergeMaps(sourceMap, overrideMap map[string]string) map[string]string {
-	result := make(map[string]string)
-	for name, value := range sourceMap {
-		result[name] = value
-	}
-	for name, value := range overrideMap {
-		result[name] = value
+// mergeMaps returns a map with the union of `source` and `override` where collisions take the value of `override`.
+func mergeMaps(source, override map[string]string) map[string]string {
+	for k, v := range override {
+		source[k] = v
```

Comment:
This is changing the functionality of the function.

Before, both of the incoming maps were left untouched and a new map was created, but this is modifying `source`.

Before:
```
source := map[string]string{"123": "abc", "345": "def"}
override := map[string]string{"123": "qwe", "678": "ghj"}
result := mergeMaps(source, override)
fmt.Printf("%+v\n", source)
// map[123:abc 345:def] (untouched)
```

After:
```
source := map[string]string{"123": "abc", "345": "def"}
override := map[string]string{"123": "qwe", "678": "ghj"}
result := mergeMaps(source, override)
fmt.Printf("%+v\n", source)
// map[123:qwe 345:def 678:ghj] (modified)
```

Reply:
Thanks for the helpful feedback. I see that Go is actually passing the pointer to `soruce` through to `mergeMaps()` which is fine, but this leads to the unintended consequence of potentially modifying`source`, I reverted the change and cleaned up the function.

Created a Go Playground to confirm: https://go.dev/play/p/wT5ltyK1KHH